build/wikia/transformers/transformWarframe.js

```diff
@@ -2,23 +2,22 @@
 
 const getColors = require('get-image-colors')
 const imageDownloader = require('image-downloader')
-
-const POLARITIES = {
-  Bar: 'naramon',
-  V: 'madurai',
-  D: 'vazarin',
-  U: 'umbra',
-  Ability: 'zenurik',
-  R: 'unairu'
-}
+const POLARITIES = require('./polarities')
 
 const transformPolarities = ({ Polarities, AuraPolarity }, targetWeapon) => {
   const outputFrame = { ...targetWeapon }
   if (AuraPolarity) {
-    outputFrame.auraPolarity = POLARITIES[AuraPolarity]
+    outputFrame.auraPolarity = (POLARITIES[AuraPolarity] || AuraPolarity || '').toLowerCase()
+    if (outputFrame.auraPolarity && !outputFrame.auraPolarity.length) outputFrame.auraPolarity = undefined
```

Comment:
Poor design in case there isn't a transformation for a polarity

i think in this case the build should fail, maybe with a more descriptive error than cannot read properties of undefined.
```suggestion
    outputFrame.auraPolarity =POLARITIES[AuraPolarity].toLowerCase()
```

Reply:
No, actually, this is in the case where we don't have a transform because we don't need one. Fallbacks by design aren't lazy if the source data is already correct. Don't need to map something that just needs a text transformation to be right. 

If it needs a mapping anyway, we don't want that to fail the whole build process, because if it's wrong, someone will notice, and the wiki has already stopped using their aliases, so the mapping is just in case they missed an old one.

Reply:
The requested change would fail on anything with an undefined mapping, which would fail builds.